platform/windows/os_windows.cpp

```diff
@@ -1828,7 +1828,9 @@ String OS_Windows::get_user_data_dir() const {
 			}
 			return get_data_path().path_join(custom_dir).replace("\\", "/");
 		} else {
-			return get_data_path().path_join(get_godot_dir_name()).path_join("app_userdata").path_join(appname).replace("\\", "/");
+			// Trim trailing periods from the user directory name as it's not valid on Windows.
+			// This check is still applied on non-Windows platforms so the project directory name is consistent across platforms.
+			return get_data_path().path_join(get_godot_dir_name()).path_join("app_userdata").path_join(appname).replace("\\", "/").rstrip(".");
```

Comment:
Won't the `custom_dir` logic above potentially still have the same problem?

I think it would be better to do this fix in `OS::get_safe_dir_name` (preventing it to end with a dot), which would make sure that both `appname` and `custom_dir` here are valid folder names.

Reply:
That makes more sense, I agree. Done :slightly_smiling_face: